File: FateGrandAutomata/GestureCompletedCallback.cs
using System.Threading;
using Android.AccessibilityServices;

namespace FateGrandAutomata
{
    class GestureCompletedCallback : AccessibilityService.GestureResultCallback
    {
        readonly ManualResetEventSlim _event;

        public GestureCompletedCallback(ManualResetEventSlim Event)
        {
            _event = Event;
        }

        public override void OnCompleted(GestureDescription GestureDescription)
        {
            _event.Set();
            base.OnCompleted(GestureDescription);
        }

        public override void OnCancelled(GestureDescription GestureDescription)
        {
            _event.Set();
            base.OnCancelled(GestureDescription);
        }
    }
}

File: app/src/main/java/com/mathewsachin/fategrandautomata/accessibility/GestureCompletedCallback.kt
package com.mathewsachin.fategrandautomata.accessibility

import android.accessibilityservice.AccessibilityService
import android.accessibilityservice.GestureDescription
import java.util.concurrent.CountDownLatch

class GestureCompletedCallback : AccessibilityService.GestureResultCallback() {
    private val latch = CountDownLatch(1)

    override fun onCompleted(gestureDescription: GestureDescription?) {
        latch.countDown()
        super.onCompleted(gestureDescription)
    }

    override fun onCancelled(gestureDescription: GestureDescription?) {
        latch.countDown()
        super.onCancelled(gestureDescription)
    }

    fun waitTillFinish() {
        latch.await()
    }
}